src/core/annotation.js

```diff
@@ -723,6 +723,38 @@ class Annotation {
     return !!(flags & flag);
   }
 
+  _buildFlags(noView, noPrint) {
```

Comment:
I've not review this method in detail, so I'm mostly trusting that you implemented all the flags correctly here :-)

Reply:
I set the F value to 0 and to 1023 in the pdf and then play with NoView/NoPrint in Acrobat, to see what bits are turned on/off, so at least it behaves the same way as in Acrobat.